openrelik_worker_common/utils.py

```diff
@@ -187,3 +189,67 @@ def create_output_file(
     """
     return OutputFile(output_path, filename, file_extension, data_type,
                       original_path, source_file_id)
+
+
+def get_path_without_root(path: str) -> str:
+    """Converts a full path to relative path without the root.
+
+    Args:
+        path: A full path.
+
+    Returns:
+        A relative path without the root.
+    """
+    # TODO(hacktobeer) Make this cross-platform
+    return str(PurePath(path).relative_to('/'))
+
+
+def build_file_tree(files: list[OutputFile]) -> tempfile.TemporaryDirectory:
+    """Creates the original file tree structure from a list of OutputFiles.
+
+    Args:
+        files: A list of OutPutFile instances.
+
+    Returns:
+        The root path of the file tree as a TemporaryDirectory.
+    """
+    os_tmpdir = tempfile.gettempdir()
+    tree_root = tempfile.TemporaryDirectory(delete=False)
```

Comment:
Using tempfile here won't work when the artifacts are stored on another disk than the OS. For example we will not be able to hard link files between and NFS mount (e.g. filestore) and OS:/tmp/ 

In the workers we create a temporary directory manually on the same path as where the artifacts are stored:

Example:
```
# Create temporary directory and hard link files for processing
temp_dir = os.path.join(output_path, uuid4().hex)
os.mkdir(temp_dir)
for input_file in input_files:
    filename = os.path.basename(input_file.get("path"))
    os.link(input_file.get("path"), f"{temp_dir}/{filename}")

... do you stuff ...

# Remove the temp_dir
shutil.rmtree(temp_dir)
```

Reply:
Makes sense. I still like a TemporaryDirectory instance to use in the delete_file_tree function but have created it with the dir parameter to make sure it is create in the OpenRelik configured output_path location.

I have also plumbed through the output_path in a non OutputFile generic way. As the run function in the worker gets this as a configured parameter.